index.js

```diff
@@ -22,7 +22,7 @@ var selfCaller = function (path, req, res, cb, url) {
     hostname: url,
     method: 'GET',
     path: path,
-    port: port || 80,
+    port: req.protocol === 'https' ? 443 : port || 80,
```

Comment:
I believe `port || req.protocol === 'https' ? 443 : 80` would make this even more robust, in case somebody uses custom port.

Reply:
I don't think that code works? I guess we can do `req.protocol === 'https' ? port || 443 : port || 80`

Reply:
Sorry, I got some wrong assumptions when testing it before, although with added parentheses it should behave as expected `port || (req.protocol === 'https' ? 443 : 80)`